x-pack/plugins/triggers_actions_ui/public/application/sections/rule_details/components/rule_event_log_data_grid.tsx

```diff
@@ -429,9 +450,27 @@ export const RuleEventLogDataGrid = (props: RuleEventLogDataGrid) => {
         isSortable: getIsColumnSortable('timed_out'),
       },
     ],
-    [getPaginatedRowIndex, onFlyoutOpen, onFilterChange, showRuleNameAndIdColumns, logs]
+    [
+      getPaginatedRowIndex,
+      onFlyoutOpen,
+      onFilterChange,
+      showRuleNameAndIdColumns,
+      showSpaceColumns,
+      logs,
+    ]
   );
 
+  useEffect(() => {
+    if (showSpaceColumns && !visibleColumns.includes('space_ids')) {
+      const ruleNameIndex = visibleColumns.findIndex((c) => c === 'rule_name');
+      const newVisibleColumns = [...visibleColumns];
+      newVisibleColumns.splice(ruleNameIndex + 1, 0, 'space_ids');
+      setVisibleColumns(newVisibleColumns);
+    } else if (!showSpaceColumns && visibleColumns.includes('space_ids')) {
+      setVisibleColumns(visibleColumns.filter((c) => c !== 'space_ids'));
+    }
+  }, [showSpaceColumns, setVisibleColumns, visibleColumns]);
```

Comment:
Curious to hear your thoughts, but I'm personally not the biggest fan of setting controlled props in `useEffects` since I got burned by it pretty hard when working with the EUI table component. 

I get what this is doing but maybe we should expect the user to have the knowledge to insert the `spaces_ids` in the visiblecolumns if they decide to show the spaces column? (if they have `showSpaceColumns === false` then we can `useMemo` the visibleColumns to remove it) 

If the intention is to do everything for the implementer of this component, maybe we can just use useMemo to add/remove the column for them internally, without syncing their `visibleColumns` variable. Just spit balling here haha